xds/internal/balancer/outlierdetection/subconn_wrapper.go

```diff
@@ -31,23 +33,54 @@ import (
 // whether or not this SubConn is ejected.
 type subConnWrapper struct {
 	balancer.SubConn
+	// The following fields are set during object creation and read-only after
+	// that.
+
 	listener func(balancer.SubConnState)
+	// healthListenerEnabled indicates whether the leaf LB policy is using a
+	// generic health listener. When enabled, ejection updates are sent via the
+	// health listener instead of the connectivity listener. Once Dualstack
+	// changes are complete, all SubConns will be created by pickfirst which
+	// uses the health listener.
+	// TODO: https://github.com/grpc/grpc-go/issues/7915 - Once Dualstack
+	// changes are complete, all SubConns will be created by pick_first and
+	// outlier detection will only use the health listener for ejection and
+	// this field can be removed.
+	healthListenerEnabled bool
 
 	// addressInfo is a pointer to the subConnWrapper's corresponding address
 	// map entry, if the map entry exists.
 	addressInfo unsafe.Pointer // *addressInfo
```

Comment:
Not sure if I brought this up earlier. But the `addressInfo` field can change during `UpdateAddresses`. So, I don't think it is correct to have this field under the read-only-after-creation group of fields.

Reply:
Move it to the top and mentioned that it's accessed atomically.